From d0f407850b60810607c679e6db3ea2ce52d19448 Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Mon, 1 Jul 2024 09:52:04 +0200
Subject: [PATCH 093/120] clk: scmi: do not register clock without a name

Avoid to register SCMI clocks with an empty name. Empty name
signifies that this clock hasn't been exposed to u-boot.

$> clk dump
 200000000      7000055         0        |-- ck_bus_bsec
 0              7000056         0        |--
 200000000      7000057         0        |-- ck_icn_p_etr

$> clk dump
 200000000      7000055         0        |-- ck_bus_bsec
 200000000      7000057         0        |-- ck_icn_p_etr

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/u-boot.git 5f86f53f52a621c00b74c1895f126ef50ea787dc clk: scmi: do not register clock without a name]
Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: Ie0a064e4502c165466040ffadb3740699ce0b66b
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/u-boot/+/390482
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Patrice CHOTARD <patrice.chotard@foss.st.com>
---
 drivers/clk/clk_scmi.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/clk/clk_scmi.c b/drivers/clk/clk_scmi.c
index cfb372e6190..2b81246fe1d 100644
--- a/drivers/clk/clk_scmi.c
+++ b/drivers/clk/clk_scmi.c
@@ -105,6 +105,10 @@ static int scmi_clk_get_attibute(struct udevice *dev, int clkid, char **name,
 		if (ret)
 			return ret;
 
+		/* Do not register clk with empty name */
+		if (!out.clock_name[0])
+			return -ECANCELED;
+
 		*name = strdup(out.clock_name);
 		*attr = out.attributes;
 	} else {
@@ -122,6 +126,10 @@ static int scmi_clk_get_attibute(struct udevice *dev, int clkid, char **name,
 		if (ret)
 			return ret;
 
+		/* Do not register clk with empty name */
+		if (!out.clock_name[0])
+			return -ECANCELED;
+
 		*name = strdup(out.clock_name);
 		*attr = out.attributes;
 	}
-- 
2.47.2

